tests/v2/validation/networking/connectivity/network_policy.go

```diff
@@ -78,3 +102,319 @@ func curlCommand(client *rancher.Client, clusterID string, url string) (string,
 	logrus.Infof("Log of the curl command curl {%v}", log)
 	return log, nil
 }
+
+// This must be a valid port number, 10250 < hostPort < 65536
+// The range 1-10250 are reserved to Rancher
+// Using a random port to avoid 'port in use' failures and allow the test to be rerun
+func getHostPort() int {
+	return rand.IntN(55283) + 10251
+}
+
+// It will allocate a port from a range 30000-32767
+// Using a random port to avoid 'port in use' failures and allow the test to be rerun
+func getNodePort() int {
+	return rand.IntN(2767) + 30000
+}
+
+// isCloudManagerEnabled is a helper function that verifies whether the cloud manager is enabled
+func isCloudManagerEnabled(client *rancher.Client, clusterID string) (bool, error) {
+	logrus.Info("Checking cluster version and if the cloud-controller-manager is installed")
+	catalogClient, err := client.GetClusterCatalogClient(clusterID)
+	if err != nil {
+		return false, err
+	}
+
+	provisioningClusterID, err := clusters.GetV1ProvisioningClusterByName(client, client.RancherConfig.ClusterName)
+	if err != nil {
+		return false, err
+	}
+
+	cluster, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ByID(provisioningClusterID)
+	if err != nil {
+		return false, err
+	}
+
+	newCluster := &provv1.Cluster{}
+	err = steveV1.ConvertToK8sType(cluster, newCluster)
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") {
+		wranglerContext, err := client.WranglerContext.DownStreamClusterWranglerContext(clusterID)
+		if err != nil {
+			return false, err
+		}
+
+		latestDaemonset, err := wranglerContext.Apps.DaemonSet().List("kube-system", metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		for _, item := range latestDaemonset.Items {
+			if strings.Contains(item.Name, "svclb-traefik") {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	}
+
+	_, err = catalogClient.Apps(kubeSystemNamespace).Get(context.TODO(), cloudControllerManager, metav1.GetOptions{})
+	if !strings.Contains(newCluster.Spec.KubernetesVersion, "k3s") && err != nil && strings.Contains(err.Error(), "not found") {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// isNodePool is a helper function that checks if the machine pool cluster size is greater than or equal to 3
+func isNodePool(steveClient *steveV1.Client) (bool, error) {
```

Comment:
question: why do we want the size of node pool to be greater than 3?
Also, would it be possible to update the name of the function to be `IsNodePoolSizeValid` or something similar isNodePool is a bit confusing.

Reply:
This test requires 3 nodes because it scales 3 deployments using the same host port